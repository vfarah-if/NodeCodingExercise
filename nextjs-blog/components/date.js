import { parseISO, format } from 'date-fns';

export default function Date({ dateString }) {
    if (dateString) {
        console.debug('datesting:', dateString);
        const date = parseISO(dateString);
        console.debug('date:', date);
        const dateFormatted = format(date, 'LLLL d, yyyy');
        console.debug('dateFormatted:', dateFormatted);
        return <time dateTime={dateString}>{dateFormatted}</time>
    } else {
        return <time>N/A</time>
    }

}