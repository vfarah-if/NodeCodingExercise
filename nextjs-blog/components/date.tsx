import { parseISO, format } from 'date-fns';

export interface DateProps {
    dateString: string
}

export default function Date({ dateString } : DateProps)  {
    if (dateString) {
        console.debug('datesting:', dateString);
        const date = parseISO(dateString);
        console.debug('date:', date);
        const dateFormatted = format(date, 'LLLL d, yyyy');
        console.debug('dateFormatted:', dateFormatted);
        return <time dateTime={dateString}>{dateFormatted}</time>
    } else {
        return <time>N/A</time>
    }

}