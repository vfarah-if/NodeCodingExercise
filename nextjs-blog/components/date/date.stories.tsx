import React from 'react';
import Date, { DateProps } from '.';
import { Story, Meta } from '@storybook/react/types-6-0';

export default { title: 'Component/Date', component: Date } as Meta;

const DateTemplate: Story<DateProps> = (args: DateProps) => (
	<>
		<Date {...args} />
	</>
);

export const EmptyDate = DateTemplate.bind({});
EmptyDate.args = {
	dateString: undefined,
};

export const DefaultDate = DateTemplate.bind({});
DefaultDate.args = {
	dateString: '2011-10-05T14:48:00.000Z',
};
