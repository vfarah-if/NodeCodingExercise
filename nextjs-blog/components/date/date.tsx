import { parseISO, format } from 'date-fns';

export interface DateProps {
    dateString: string
}

export default function Date({ dateString } : DateProps)  {
    if (dateString) {
        console.debug('dateString:', dateString);
        const date = parseISO(dateString);
        console.debug('date:', date);
        const dateFormatted = format(date, 'LLLL d, yyyy');
        console.debug('dateFormatted:', dateFormatted);
        return <time dateTime={dateString}>{dateFormatted}</time>
    } else {
        return <time>Infinite and no time assigned</time>
    }

}