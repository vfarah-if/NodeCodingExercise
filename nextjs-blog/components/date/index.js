import Date, { DateProps } from './date';
export default Date;