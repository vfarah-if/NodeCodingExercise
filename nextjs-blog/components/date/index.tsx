export { default } from './date';
export type { DateProps } from './date';

