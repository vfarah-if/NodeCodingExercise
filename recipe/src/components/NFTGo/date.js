const MILLISECONDS_PER_SECOND = 1000;
// const SECONDS_PER_MINUTE = 60;
// const MINUTES_PER_HOUR = 60;
// const HOURS_PER_DAY = 24;
// const DAYS_PER_WEEK = 7;
// const MONTHS_PER_YEAR = 12;

// const SECOND = MILLISECONDS_PER_SECOND;
// const MINUTE = SECOND * SECONDS_PER_MINUTE;
// const HOUR = MINUTE * MINUTES_PER_HOUR;
// const DAY = HOUR * HOURS_PER_DAY;
// const WEEK = DAY * DAYS_PER_WEEK;
// const YEAR = DAY * 365.24;
// const NORMAL_YEAR = DAY * 365;
// const LEAP_YEAR = DAY * 366;
// const DECADE = 10 * YEAR;
// const HALF_YEAR = YEAR / 2;
// const AVERAGE_MONTH = YEAR / 12;

export const subtractDays = (date, days) => {
  const result = new Date(date);
  result.setDate(result.getDate() - days);
  return result;
};

export const epochToDate = (epochTimeStamp) => {
  return new Date(epochTimeStamp * MILLISECONDS_PER_SECOND);
};

export const dateToEpoch = (date) => {
  return (date.getTime() - date.getMilliseconds()) / MILLISECONDS_PER_SECOND;
};

export const secondsSinceEpoch = (numericDateValue) => {
  return Math.floor(numericDateValue / MILLISECONDS_PER_SECOND);
};
