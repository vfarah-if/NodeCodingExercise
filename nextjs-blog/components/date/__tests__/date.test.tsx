import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { EmptyDate, DefaultDate } from '../date.stories';

describe('Date tests as defined by stories', () => {
	test('should generate a no time assigned message when nothing assigned', () => {
		const component = render(<EmptyDate />);

		expect(component.container).toMatchSnapshot();
	});

	test('should generate a default date when assigned all expected values', () => {
		const component = render(<DefaultDate dateString={DefaultDate.args.dateString} />);

		expect(component.container).toMatchSnapshot();
	});
});
