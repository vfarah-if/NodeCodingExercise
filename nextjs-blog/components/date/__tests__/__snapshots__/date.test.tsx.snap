// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date tests as defined by stories should generate a default date when assigned all expected values 1`] = `
<div>
  <time
    datetime="2011-10-05T14:48:00.000Z"
  >
    October 5, 2011
  </time>
</div>
`;

exports[`Date tests as defined by stories should generate a no time assigned message when nothing assigned 1`] = `
<div>
  <time>
    Infinite and no time assigned
  </time>
</div>
`;
